Add a Basic Anaglyph Node

danieldresser-ie opened the pull request:
This is a very basic anaglyph node, without any controls, but it should work for viewing multi-view images with left and right views, which seems like all we need for the moment.

Also includes a fix for a bug with Merge that turned up while testing this.

johnhaddon commented on python/GafferImage/Anaglyph.py:
```diff
@@ -0,0 +1,70 @@
+
+		self["__DeleteChannelsLeft"] = GafferImage.DeleteChannels()
+		self["__DeleteChannelsLeft"]["in"].setInput( self["__SelectLeft"]["out"] )
+		self["__DeleteChannelsLeft"]["channels"].setValue( '[GB]' )
```
If we also deleted `*.[GB]` here and `*.[R]` for the right view, this node would work for additional layers too.

johnhaddon commented on python/GafferImage/Anaglyph.py:
```diff
@@ -0,0 +1,70 @@
+		self["__Merge"]["operation"].setValue( GafferImage.Merge.Operation.Max )
+
+		self['out'].setFlags(Gaffer.Plug.Flags.Serialisable, False)
+		self["out"].setInput( self["__Merge"]["out"] )
```
We need to hook the output up via a switch so the node can be disabled.

johnhaddon commented on python/GafferImage/Anaglyph.py:
```diff
@@ -0,0 +1,70 @@
+
+class Anaglyph( GafferImage.ImageProcessor ) :
+
+	def __init__(self, name = 'Anaglpyh' ) :
```
Typo - should be `"Anaglyph"`.

johnhaddon reviewed (changes requested):
Thanks Daniel. In addition to the few comments inline, we also need to :

- [ ] Update Changes.md
- [ ] Add AnaglyphUI.py with a node description
- [ ] Add a basic unit test of some sort. At a minimum I suggest we should check that just a default view comes out, with the appropriate channel names (perhaps including custom layers, to prove the suggestion I made works).

Cheers...
John

danieldresser-ie commented:
Feedback addressed.  Only thing that came up while adding the test was that I again needed the ignoreDataWindow flag of assertImagesEqual, which I had needed several times while testing multiview.  I thought it might be time to finally add it ... then I took a look, and saw that I had already done that a while ago, although it didn't seem to be quite working.  So I fixed it, used it in the Anaglyph test, and also went back and cleaned up the multiview image reader/writer tests using it.

johnhaddon commented:
I want to get this merged this morning so I've pushed a rebase which fixes the ImageWriter test failure and removes the unused debug ImageWriter from AnaglyphTest.
